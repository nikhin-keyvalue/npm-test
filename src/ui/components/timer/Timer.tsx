import Preact, { h } from 'preact';

import CircleCountDown from '../sektor';
import { DEFAULT_PAYMENT_TIME_LIMIT, DEFAULT_PAYMENT_TIME_REMAINING } from 'constants/common';

import './style.scss';

interface TimerProps {
  //duration in millisceonds
  totalTimeDuration?: number;
  currentTime?: number;
  timerText?: string;
  startTimer: boolean;
  //prop used to adjust style issue in certain components
  styleOffset?: boolean;
}
const Timer: Preact.FunctionComponent<TimerProps> = ({
  totalTimeDuration = DEFAULT_PAYMENT_TIME_LIMIT,
  currentTime = DEFAULT_PAYMENT_TIME_REMAINING,
  timerText,
  startTimer,
  styleOffset = false
}) => {
  const formatTime = (time: number) => {
    const minutes: number = Math.floor(time / 60);
    let seconds: number | string = time % 60;

    if (seconds < 10) seconds = `0${seconds}`;

    return `${minutes}:${seconds}`;
  };

  return (
    <div className='zamp-timer-wrapper'>
      <div className='zamp-border-circle'>
        <div className={`zamp-sektor-wrapper ${styleOffset && 'zamp-sektor-offset'}`}>
          <CircleCountDown time={totalTimeDuration} currentTime={currentTime} startTimer={startTimer} />
        </div>
      </div>
      <div className='zamp-time-wrapper'>{`${formatTime(currentTime)} ${timerText}`}</div>
    </div>
  );
};

Timer.defaultProps = {
  timerText: ''
};

export default Timer;
