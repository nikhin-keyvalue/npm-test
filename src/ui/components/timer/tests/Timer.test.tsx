import { h } from 'preact';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Adapter } from 'enzyme-adapter-preact-pure';

import Timer from '../Timer';

configure({ adapter: new Adapter() });

describe('Counter', () => {
  it('render components', () => {
    const wrapper = shallow(<Timer totalTimeDuration={0} startTimer={false} />);
    const tree = shallowToJson(wrapper);

    expect(tree).toMatchSnapshot();
  });

  // TODO: setInterval test cases to be explored
});
