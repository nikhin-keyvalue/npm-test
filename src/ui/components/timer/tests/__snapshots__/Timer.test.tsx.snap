// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render components 1`] = `
<div
  className="zamp-timer-wrapper"
>
  <div
    className="zamp-border-circle"
  >
    <div
      className="zamp-sektor-wrapper false"
    >
      <CircleCountDown
        currentTime={0}
        startTimer={false}
        time={0}
      />
    </div>
  </div>
  <div
    className="zamp-time-wrapper"
  >
    0:00 
  </div>
</div>
`;
